src/wallet/walletdb.cpp

```diff
@@ -646,20 +646,17 @@ DBErrors CWalletDB::LoadWallet(CWallet* pwallet)
     return result;
 }
 
-DBErrors CWalletDB::FindWalletTx(CWallet* pwallet, vector<uint256>& vTxHash, vector<CWalletTx>& vWtx)
+DBErrors CWalletDB::FindWalletTx(vector<uint256>& vTxHash, vector<CWalletTx>& vWtx)
 {
-    pwallet->vchDefaultKey = CPubKey();
```

Comment:
That probably should have been moved to ZapWalletTx. It wiped the default key, to make sure a new key would be generated next time. The default key is effectively unused currently, we should remove it entirely.